launch/publish_demo_data.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
+"""
+Periodically publish gps positions in a circle around a given origin as NavSatFix messages.
+"""
+import rclpy
+import math
+import numpy as np
+from argparse import ArgumentParser
+from rclpy.time import CONVERSION_CONSTANT
+from sensor_msgs.msg import NavSatFix, NavSatStatus
+
+
+class CircularTranslate():
+    """Translate fix periodically in a circle"""
+
+    def update(self, t, fix):
+        r = 0.001
+        circle_period = 10.0
+        t_sec = t.nanoseconds / CONVERSION_CONSTANT
+        phi = math.tau * (t_sec / circle_period)
+        fix.latitude += math.cos(phi) * r
+        fix.longitude += math.sin(phi) * r
+
+
+class ApplyNoise():
+    """Translate fix position using gaussian noise"""
+    def __init__(self, stddev):
+        self.stddev = stddev
+
+    def update(self, t, fix):
+        noise = np.random.normal(0, self.stddev, 2)
+        fix.latitude += noise[0]
+        fix.longitude += noise[1]
+
+
+class LooseFix():
+    """Set NavSatFix status to NO_FIX periodically"""
+    def __init__(self, phase, period):
+        self.phase = phase
+        self.period = period
+
+    def update(self, t, fix):
+        t_sec = t.nanoseconds / CONVERSION_CONSTANT
```

Comment:
ROS2 Time doesn't have a seconds property or function? Weird...

Reply:
I haven't seen it calculated that way. Most of the time I grab the first item in the tuple. But this does work.

```
In [8]: t
Out[8]: Time(nanoseconds=1676676204306391161, clock_type=ROS_TIME)

In [9]: t.seconds_nanoseconds()
Out[9]: (1676676204, 306391161)
```